### DISCUSSION TITLE
How do you pass Form fields containing commas ?

### DISCUSSION THREAD

**Discussion by euri10** [fastapi/fastapi#8210] [Questions] [answered]
**Description**

How can I pass form fields that contain commas ? In the below snippet I used the form field `hub_callback` but in real life I'd like it to be `hub.callback`

```
@app.post("/subscribe")
async def subscribe(*, hub_callback: str = Form(...)):
    return {"hub_callback": hub_callback}
```
the below curl is valid, question is how can therefore can  have a python variable in the router that is different from the "generated" form field value ?

`curl -X POST "https://editor.swagger.io/subscribe" -H "accept: application/json" -H "Content-Type: application/x-www-form-urlencoded" -d "hub.callback=test"`

**Comment by wshayes** [comment, marked as answer]
Couldn't you use pydantic aliasing for this?

**Comment by euri10** [comment]
Seems like indeed this generates the correct documentation @wshayes :
```
@app.post("/subscribe")
async def subscribe(*, hub_callback: str = Form(..., alias="hub.callback")):
    return {"hub_callback": hub_callback}
```

however I got a 400 error when passing a string to the form...not that useful a message
`{
  "detail": "There was an error parsing the body"
}`

**Comment by vitalik** [comment]
check the console logs - most likely you need to install python-multipart ;)

**Comment by euri10** [comment]
that's exactly it thanks @vitalik

**Comment by tiangolo** [comment]
Thanks for the help @wshayes and @vitalik ! :cake: :rocket: 

I'm glad you solved it @euri10 , thanks for reporting back and closing it :tada: :heavy_check_mark:
